docs/shared/checkpoint_learn_serialize.md

```diff
@@ -0,0 +1,118 @@
+\page ckpt_learn_serialize How to Serialize Data
+
+*Serialization* is the process of converting an object into a simple format
+that can be stored or transmitted and reconstructed later.
+checkpoint translates the object into a set of contiguous bits
+and provides the steps to reverse the process, i.e. to reconstitute
+the object from the set of bits.
+
+The extraction of the data from a series of bytes is called
+*deserialization*.
+
+\section serialize_builtin Serialization of built-in types
+
+This action is straightforward with the ```|``` operator.
+
+When the variable `a` has a built-in type and it needs to be serialized
+into the serializer object `s`, we simply write
+
+\code{.cpp}
+s | a;
+\endcode
+
+\section serialize_stl Serialization of STL containers
+
+The `|` operator has been overloaded for the STL containers:
+`std::array`, `std::deque`, `std::list`,
+`std::map`, `std::multimap`, `std::multiset`,
+`std::queue`,
+`std::set`, `std::string`,
+`std::unordered_map`, `std::unordered_multimap`,
+`std::unordered_multiset`, `std::unordered_set`,
+`std::vector`.
+
+The `|` operator is also overloaded for `std::pair`, `std::shared_ptr`, `std::unique_ptr`.
```

Comment:
Is `std::tuple` supported? If not, it should be.

Reply:
`checkpoint/container/tuple_serialize.h` looks legit, it should be added to the list

Reply:
Added to docs